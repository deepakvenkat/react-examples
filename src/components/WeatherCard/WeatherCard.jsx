import React from 'react';
import {
  Card,
  CardContent,
  Typography,
  Button,
} from '@material-ui/core';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

import WeatherIcon from '../WeatherIcon';
const styles = {
  root: {
    width: '125px',
    height: '200px',
    textAlign: 'center',
  },
  temp: {
    display: 'inline-block',
  },
};

const WeatherCard = ({
  classes,
  day,
  high,
  low,
  weather,
}) =>  (
  <Card
    classes={classes}
  >
    <CardContent>
      <Typography className={classes.title} color="textSecondary" gutterBottom>
        {day}
      </Typography>
      <WeatherIcon weather={weather} />
      <br />
      <Typography component="p" className={classes.temp}>
        {high}/
      </Typography>
      <Typography color="textSecondary" component="p" className={classes.temp}>
        {low}
      </Typography>
    </CardContent>
  </Card>
);

WeatherCard.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  day: PropTypes.string.isRequired,
  high: PropTypes.string.isRequired,
  low: PropTypes.string.isRequired,
  weather: PropTypes.string.isRequired,
};

export default withStyles(styles)(WeatherCard);
