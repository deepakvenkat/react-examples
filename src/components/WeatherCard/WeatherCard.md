## Weather Card

```jsx
<WeatherCard
  day="Wed"
  high="77"
  low="63"
  weather="sunny"
/>
```
