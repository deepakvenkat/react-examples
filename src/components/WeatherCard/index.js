import WeatherCard from './WeatherCard.jsx';

export default WeatherCard;
