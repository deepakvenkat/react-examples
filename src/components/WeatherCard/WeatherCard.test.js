import React from 'react';
import { render } from '@testing-library/react';
import WeatherCard from './WeatherCard';

test('renders the WeatherCard', () => {
  const { asFragment } = render(
    <WeatherCard
      weather="sunny"
      day="Wed"
      high="77"
      low="63"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
