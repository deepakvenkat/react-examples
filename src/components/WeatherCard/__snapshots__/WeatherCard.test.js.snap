// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the WeatherCard 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root WeatherCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
      >
        Wed
      </p>
      <div
        class="MuiAvatar-root MuiAvatar-square WeatherIcon-avatar-4 MuiAvatar-colorDefault"
      >
        <div
          class="WeatherIcon-first-3 WeatherIcon-first-5"
        />
      </div>
      <br />
      <p
        class="MuiTypography-root WeatherCard-temp-2 MuiTypography-body1"
      >
        77/
      </p>
      <p
        class="MuiTypography-root WeatherCard-temp-2 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        63
      </p>
    </div>
  </div>
</DocumentFragment>
`;
